subsys/net/ip/promiscuous.c

```diff
@@ -0,0 +1,88 @@
+/** @file
+ * @brief Promiscuous mode support
+ *
+ * Allow user to receive all network packets that are seen by a
+ * network interface. This requires that network device driver
+ * supports promiscuous mode.
+ */
+
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_DEBUG_PROMISC)
+#define SYS_LOG_DOMAIN "net/promisc"
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <kernel.h>
+#include <errno.h>
+
+#include <net/net_if.h>
+#include <net/net_core.h>
+
+static K_FIFO_DEFINE(promiscuous_queue);
+static atomic_t enabled = ATOMIC_INIT(0);
+
+struct net_pkt *net_promisc_mode_wait_data(s32_t timeout)
+{
+	return k_fifo_get(&promiscuous_queue, timeout);
+}
+
+int net_promisc_mode_on(struct net_if *iface)
+{
+	if (!iface) {
+		return -EINVAL;
+	}
+
+	if (!atomic_cas(&enabled, 0, 1)) {
```

Comment:
ok works on one iface at a time. Fine by me, let's see if we get the use case for 2+ iface at some point

Reply:
Hmm, you are right. Actually it was meant to work with multiple interfaces as seen by the sample app. Let's see if it would be easy to add that support.